src/main/java/dansplugins/factionsystem/commands/abs/SubCommand.java

```diff
@@ -0,0 +1,311 @@
+     * @param names of the command, for example, "Fly, FFly, Flight".
+     * @param playerCommand if the command is exclusive to players.
+     * @param requiresFaction if the command requires a Faction to perform.
+     * @param requiresOfficer if the command requires officer or higher.
+     * @param requiresOwner if the command is reserved for Owners.
+     */
+    public SubCommand(String[] names,
+                      boolean playerCommand,
+                      boolean requiresFaction,
+                      boolean requiresOfficer,
+                      boolean requiresOwner) {
+        // Local variables standing for instances of constantly used instances.
+        this.locale = LocaleManager.getInstance();
+        this.data = PersistentData.getInstance();
+        this.parser = ArgumentParser.getInstance();
+        this.ephemeral = EphemeralData.getInstance();
+        this.chunks = ChunkManager.getInstance();
+        this.dynmap = DynmapManager.getInstance();
+        this.configManager = ConfigManager.getInstance();
+
+        // Load Command Names.
+        this.names = new String[names.length];
+        for (int i = 0; i < this.names.length; i++) {
+            String name = names[i];
+            if (name.contains(LOCALE_PREFIX)) name = locale.getText(name.replace(LOCALE_PREFIX, ""));
+            this.names[i] = name;
+        }
+
+        this.playerCommand = playerCommand;
+        this.requiresFaction = requiresFaction;
+        this.requiresOfficer = requiresOfficer;
+        this.requiresOwner = requiresOwner;
+    }
+
+    /**
+     * Constructor to initialise a command without owner/faction checks.
+     * @param names of the command.
+     * @param playerCommand if the command is exclusive to players.
+     * @param requiresFaction if the command requires a Faction to do.
+     */
+    public SubCommand(String[] names, boolean playerCommand, boolean requiresFaction) {
+        this(names, playerCommand, requiresFaction, false, false);
+    }
+
+    /**
+     * Constructor to initialise a command without faction checks.
+     * @param names of the command.
+     * @param playerCommand if the command is exclusive to players.
+     */
+    public SubCommand(String[] names, boolean playerCommand) {
+        this(names, playerCommand, false);
+    }
+
+    /**
+     * Method to be called by the command interpreter <em>only</em>.
+     * <p>
+     *     This method uses the in-class variables to call a different method based on the parameters specified.
+     *     <br>For example, if {@link SubCommand#playerCommand} is {@code true},
+     *     <br>{@link SubCommand#execute(Player, String[], String)} is executed,
+     *     <br>not {@link SubCommand#execute(CommandSender, String[], String)}.
+     * </p>
+     * @param sender who sent the command.
+     * @param args of the command.
+     * @param key of the sub-command.
+     */
+    public void performCommand(CommandSender sender, String[] args, String key) {
```

Comment:
I think this method is excellent! I've always felt like doing something like this.